projects/CondSeg/README.md

```diff
@@ -1,2 +1,108 @@
+
+2. **Configuration Management**: The `config.py` file and YAML files in the `configs` directory allow for flexible and extensive configuration of model parameters, training settings, and dataset specifics. 
+
+3. **Model Training and Inference**: The `trainer.py` file provides the `TrainerCondSeg` class, encapsulating the logic for setting up the model, training the model, conducting validation, and performing inference. It integrates the model with the data loaders, handles the training loop, loss computation, and implements various functionalities like checkpoint saving and learning rate scheduling.
+
+4. **Execution**: `main.py` serves as the starting point for running the model. It sets up the device configuration, initializes the trainer, and triggers either the training or inference process based on the provided command-line arguments.
+
+## Usage
+
+To use this repository:
+
+1. **Setup**: Setup up PyTorch-Connetomics as specified in the main README.
+
+2. **Data Preparation**: Organize your volumetric data and corresponding labels in the required format and structure.
+
+3. **Configuration**: Adjust the parameters in the provided YAML configuration files in the `configs` directory to suit your specific requirements.
+
+4. **Run**: Execute `main.py` with the desired mode (training or inference) and other necessary command-line arguments. 
+
+5. **Results**: Output, such as trained models and inference results, will be saved in the specified output directories.
+
+## Data
+
+### Type of Data Required
+
+1. **Volumetric Data**: The primary type of data required is volumetric data, which are essentially 3D images or volumes. These could be medical scans like MRI or CT images, 3D microscopy images, or any other form of 3D data that requires segmentation.
+
+2. **Labels/Annotations**: For supervised learning, you'll need corresponding labels or annotations for your volumetric data. These labels should indicate the segments or regions of interest in the volumes. In conditional segmentation, these labels might also include specific conditions or attributes for each segment.
+
+### Data Format
+
+- The data (both images and labels) should be in the `.h5`  (HDF5) or the `.tif` (TIFF) file format to be compatible with the tools used in the repository.
+- While the system handles certain normalization and preprocessing steps, users should still perform a thorough inspection and preprocessing of their data to ensure compatibility and optimal processing within the system.
+
+### Data Structure
+
+- The system can handle both single complete volumes and subvolumes (chunks). This is configured by the `DATA_CHUNK_NUM` parameter in the YAML configuration files.
+- For inference, there’s a distinction between processing volumes "singly" or in "chunks," as indicated by the `INFERENCE.DO_SINGLY` configuration parameter. The choice between these options would depend on the size of your data and available computational resources.
```

Comment:
@zudi-lin did I interpret this correctly and does it mean that the setup is also able to work on an instance basis?